test/test_basic_logic.py

```diff
@@ -439,8 +439,16 @@ def test_dict_headers(self):
         """
         Sending headers using dict is deprecated but still valid.
         """
+        # One of the steps in the outbound header validation logic checks
+        # that we don't send a pseudo-header after we've sent a regular
+        # header.  We can't guarantee that if you send headers as a dict,
+        # because dicts are unordered.
+        config = h2.config.H2Configuration(
+            validate_outbound_headers=False
+        )
+
```

Comment:
I think I'm ok with doing this prior to 3.0. As with all the validation checks, it can be disabled, and issuing ill-formed header blocks is a bug that should be alerted.